File: 485-max-consecutive-ones/max-consecutive-ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_count = 0
        curr_count = 0
        
        for num in nums:
            if num == 1:
                curr_count += 1
                max_count = max(max_count, curr_count)
            else:
                curr_count = 0
        return max_count

File: 485-max-consecutive-ones/max-consecutive-ones.c
int findMaxConsecutiveOnes(int* nums, int numsSize) {
    int max_count = 0;
    int count = 0;
    
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == 1) {
            count++;
            if (count > max_count) {
                max_count = count;
            }
        }
        else {
            count = 0;
        }
    }
    return (max_count);
}